src/eleveldb.erl

```diff
@@ -395,105 +411,280 @@ validate_type(_, _)                                          -> false.
-
-destroy_test() -> [{destroy_test_Z(), l} || l <- lists:seq(1, 20)].
-destroy_test_Z() ->
-    os:cmd("rm -rf /tmp/eleveldb.destroy.test"),
-    {ok, Ref} = open("/tmp/eleveldb.destroy.test", [{create_if_missing, true}]),
-    ok = ?MODULE:put(Ref, <<"def">>, <<"456">>, []),
-    {ok, <<"456">>} = ?MODULE:get(Ref, <<"def">>, []),
-    close(Ref),
-    ok = ?MODULE:destroy("/tmp/eleveldb.destroy.test", []),
-    {error, {db_open, _}} = open("/tmp/eleveldb.destroy.test", [{error_if_exists, true}]).
-
-compression_test() -> [{compression_test_Z(), l} || l <- lists:seq(1, 20)].
-compression_test_Z() ->
-    CompressibleData = list_to_binary([0 || _X <- lists:seq(1,20)]),
-    os:cmd("rm -rf /tmp/eleveldb.compress.0 /tmp/eleveldb.compress.1"),
-    {ok, Ref0} = open("/tmp/eleveldb.compress.0", [{write_buffer_size, 5},
-                                                   {create_if_missing, true},
-                                                   {compression, false}]),
-    [ok = ?MODULE:put(Ref0, <<I:64/unsigned>>, CompressibleData, [{sync, true}]) ||
-        I <- lists:seq(1,10)],
-    {ok, Ref1} = open("/tmp/eleveldb.compress.1", [{write_buffer_size, 5},
-                                                   {create_if_missing, true},
-                                                   {compression, true}]),
-    [ok = ?MODULE:put(Ref1, <<I:64/unsigned>>, CompressibleData, [{sync, true}]) ||
-        I <- lists:seq(1,10)],
-	%% Check both of the LOG files created to see if the compression option was correctly
-	%% passed down
-	MatchCompressOption =
-		fun(File, Expected) ->
-				{ok, Contents} = file:read_file(File),
-				case re:run(Contents, "Options.compression: " ++ Expected) of
-					{match, _} -> match;
-					nomatch -> nomatch
-				end
-		end,
-	Log0Option = MatchCompressOption("/tmp/eleveldb.compress.0/LOG", "0"),
-	Log1Option = MatchCompressOption("/tmp/eleveldb.compress.1/LOG", "1"),
-	?assert(Log0Option =:= match andalso Log1Option =:= match).
-
-
-close_test() -> [{close_test_Z(), l} || l <- lists:seq(1, 20)].
-close_test_Z() ->
-    os:cmd("rm -rf /tmp/eleveldb.close.test"),
-    {ok, Ref} = open("/tmp/eleveldb.close.test", [{create_if_missing, true}]),
-    ?assertEqual(ok, close(Ref)),
-    ?assertEqual({error, einval}, close(Ref)).
-
-close_fold_test() -> [{close_fold_test_Z(), l} || l <- lists:seq(1, 20)].
-close_fold_test_Z() ->
-    os:cmd("rm -rf /tmp/eleveldb.close_fold.test"),
-    {ok, Ref} = open("/tmp/eleveldb.close_fold.test", [{create_if_missing, true}]),
-    ok = eleveldb:put(Ref, <<"k">>,<<"v">>,[]),
-    ?assertException(throw, {iterator_closed, ok}, % ok is returned by close as the acc
-                     eleveldb:fold(Ref, fun(_,_A) -> eleveldb:close(Ref) end, undefined, [])).
+%% ===================================================================
+%% Exported Test Helpers
+%% ===================================================================
+
+-spec assert_close(DbRef :: db_ref()) -> ok | no_return().
```

Comment:
I use a single `%` prefix when it's not published in edoc.
If it doesn't have a `%% @doc` comment, it's because it's not exported in all cases, and edoc will blow up on it.  Basically, anything within the `-ifdef(TEST).` block with any edoc comments causes failure (edoc's not very smart sometimes).